### ISSUE TITLE
Elemental fails to build, Could not determine BLAS format

### ISSUE THREAD

**Issue by kwkelly** [Homebrew/homebrew-science#2107]
During configuration step, cmake reports that it can not determine the BLAS format.

```
brew install -v elemental
==> Installing elemental from homebrew/homebrew-science
/usr/local/bin/mpicc --version
/usr/local/bin/mpicxx --version
/usr/local/bin/mpif90 --version
/usr/local/bin/mpicc --version
/usr/local/bin/mpicxx --version
/usr/local/bin/mpif90 --version
/usr/local/bin/mpicc --version
/usr/local/bin/mpicxx --version
/usr/local/bin/mpif90 --version
==> Downloading http://libelemental.org/pub/releases/Elemental-0.84-p1.tgz
Already downloaded: /Library/Caches/Homebrew/elemental-0.84-p1.tgz
==> Verifying elemental-0.84-p1.tgz checksum
tar xf /Library/Caches/Homebrew/elemental-0.84-p1.tgz
==> cmake .. -DCMAKE_INSTALL_PREFIX=/usr/local/Cellar/elemental/0.84-p1 -DCMAKE_BUILD_TYPE=PureRelease -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_VERBOSE_MAKEFILE=ON -Wno-dev -DELEM_TESTS=ON -DELEM_EXAMPLES=ON
-- The C compiler identification is AppleClang 6.1.0.6020049
-- The CXX compiler identification is AppleClang 6.1.0.6020049
-- Check for working C compiler: /usr/local/Library/ENV/4.3/clang
-- Check for working C compiler: /usr/local/Library/ENV/4.3/clang -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/local/Library/ENV/4.3/clang++
-- Check for working CXX compiler: /usr/local/Library/ENV/4.3/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- The Fortran compiler identification is GNU
-- Checking whether Fortran compiler has -isysroot
-- Checking whether Fortran compiler has -isysroot - yes
-- Checking whether Fortran compiler supports OSX deployment target flag
-- Checking whether Fortran compiler supports OSX deployment target flag - yes
-- Check for working Fortran compiler: /usr/local/bin/gfortran
-- Check for working Fortran compiler: /usr/local/bin/gfortran  -- works
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Checking whether /usr/local/bin/gfortran supports Fortran 90
-- Checking whether /usr/local/bin/gfortran supports Fortran 90 -- yes
-- Detecting Fortran/C Interface
-- Detecting Fortran/C Interface - Found GLOBAL and MODULE mangling
-- Verifying Fortran/CXX Compiler Compatibility
-- Verifying Fortran/CXX Compiler Compatibility - Success
-- Performing Test _HAS_CXX11_FLAG
-- Performing Test _HAS_CXX11_FLAG - Success
-- Checking C++ support for "auto"
-- Checking C++ support for "auto": works
-- Checking C++ support for "class_override_final"
-- Checking C++ support for "class_override_final": works
-- Checking C++ support for "constexpr"
-- Checking C++ support for "constexpr": works
-- Checking C++ support for "cstdint_header"
-- Checking C++ support for "cstdint_header": works
-- Checking C++ support for "decltype"
-- Checking C++ support for "decltype": works
-- Checking C++ support for "defaulted_functions"
-- Checking C++ support for "defaulted_functions": works
-- Checking C++ support for "delegating_constructors"
-- Checking C++ support for "delegating_constructors": works
-- Checking C++ support for "deleted_functions"
-- Checking C++ support for "deleted_functions": works
-- Checking C++ support for "func_identifier"
-- Checking C++ support for "func_identifier": works
-- Checking C++ support for "initializer_list"
-- Checking C++ support for "initializer_list": works
-- Checking C++ support for "lambda"
-- Checking C++ support for "lambda": works
-- Checking C++ support for "long_long"
-- Checking C++ support for "long_long": works
-- Checking C++ support for "nullptr"
-- Checking C++ support for "nullptr": works
-- Checking C++ support for "rvalue_references"
-- Checking C++ support for "rvalue_references": works
-- Checking C++ support for "sizeof_member"
-- Checking C++ support for "sizeof_member": works
-- Checking C++ support for "static_assert"
-- Checking C++ support for "static_assert": works
-- Checking C++ support for "variadic_templates"
-- Checking C++ support for "variadic_templates": works
-- Found CXXFeatures: TRUE
-- CXX11_COMPILER_FLAGS=-std=c++11
-- Performing Test ELEM_HAVE_TEMPLATE_ALIAS
-- Performing Test ELEM_HAVE_TEMPLATE_ALIAS - Success
-- Performing Test ELEM_HAVE_STEADYCLOCK
-- Performing Test ELEM_HAVE_STEADYCLOCK - Success
-- Performing Test ELEM_HAVE_NOEXCEPT
-- Performing Test ELEM_HAVE_NOEXCEPT - Success
-- Performing Test ELEM_HAVE_NORMAL_DIST
-- Performing Test ELEM_HAVE_NORMAL_DIST - Success
-- Performing Test ELEM_HAVE_UNIFORM_INT_DIST
-- Performing Test ELEM_HAVE_UNIFORM_INT_DIST - Success
-- Performing Test ELEM_HAVE_UNIFORM_REAL_DIST
-- Performing Test ELEM_HAVE_UNIFORM_REAL_DIST - Success
-- Performing Test HAVE___restrict__
-- Performing Test HAVE___restrict__ - Success
-- Performing Test HAVE___restrict
-- Performing Test HAVE___restrict - Success
-- Performing Test HAVE_restrict
-- Performing Test HAVE_restrict - Failed
-- Using __restrict__ keyword.
-- Found MPI_C: /usr/local/lib/libmpi.dylib
-- Found MPI_CXX: /usr/local/lib/libmpi_cxx.dylib;/usr/local/lib/libmpi.dylib
-- Found MPI_Fortran: /usr/local/lib/libmpi_usempif08.dylib;/usr/local/lib/libmpi_usempi_ignore_tkr.a;/usr/local/lib/libmpi_mpifh.dylib;/usr/local/lib/libmpi.dylib
-- Looking for MPI_Reduce_scatter
-- Looking for MPI_Reduce_scatter - found
-- Looking for MPI_Reduce_scatter_block
-- Looking for MPI_Reduce_scatter_block - found
-- Looking for MPI_Iallgather
-- Looking for MPI_Iallgather - found
-- Looking for MPIX_Iallgather
-- Looking for MPIX_Iallgather - not found
-- Looking for MPI_Init_thread
-- Looking for MPI_Init_thread - found
-- Looking for MPI_Query_thread
-- Looking for MPI_Query_thread - found
-- Looking for MPI_Comm_set_errhandler
-- Looking for MPI_Comm_set_errhandler - found
-- Looking for MPI_Type_create_struct
-- Looking for MPI_Type_create_struct - found
-- Performing Test ELEM_HAVE_MPI_IN_PLACE
-- Performing Test ELEM_HAVE_MPI_IN_PLACE - Success
-- Performing Test ELEM_HAVE_MPI_LONG_LONG
-- Performing Test ELEM_HAVE_MPI_LONG_LONG - Success
-- Performing Test ELEM_HAVE_MPI_COMM_F2C
-- Performing Test ELEM_HAVE_MPI_COMM_F2C - Success
-- Performing Test ELEM_HAVE_OMP_COLLAPSE
-- Performing Test ELEM_HAVE_OMP_COLLAPSE - Failed
-- Using Apple vecLib framework.
-- Looking for daxpy
-- Looking for daxpy - not found
-- Looking for daxpy_
-- Looking for daxpy_ - not found
-- Looking for dpotrf
-- Looking for dpotrf - not found
-- Looking for dpotrf_
-- Looking for dpotrf_ - not found
CMake Error at cmake/tests/Math.cmake:53 (message):
  Could not determine BLAS format.
Call Stack (most recent call first):
  CMakeLists.txt:209 (include)


-- Configuring incomplete, errors occurred!
See also "/tmp/elemental20150415-21151-ilqr4z/Elemental-0.84-p1/build/CMakeFiles/CMakeOutput.log".
See also "/tmp/elemental20150415-21151-ilqr4z/Elemental-0.84-p1/build/CMakeFiles/CMakeError.log".
```

Curiously, the files referenced at the end did not exist. Using `--HEAD` made it thorugh install, but there were linking errors due to a conflict with metis.

```
The formula built, but is not symlinked into /usr/local
Could not symlink bin/cmpfillin
Target /usr/local/bin/cmpfillin
is a symlink belonging to metis.
```

**Comment by dpo** [comment]
I'm working on an upgrade to version 0.85 that links against Homebrew `metis`.

**Comment by dpo** [comment]
https://github.com/elemental/Elemental/issues/55

**Comment by dpo** [comment]
Fixed in #2158.
